### PR TITLE
Add thanos query frontend and remove the cortex one

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <yb532204897@gmail.com>

### CODE REVIEW

File: components/observatorium-configure.libsonnet [hunk 1]
```diff
@@ -35,8 +35,8 @@
     config+:: obs.config.query,
   },
 
-  queryCache+:: {
-    config+:: obs.config.queryCache,
+  queryFrontend+:: {
```
Comment (kakkoyun) [inline components/observatorium-configure.libsonnet:38]: Probably operator needs an update https://github.com/observatorium/deployments/blob/1964b4e5e797c8e77a86bc17c61ba855c35dfad8/operator/jsonnet/operator-config.libsonnet#L55

Comment (yeya24) [reply components/observatorium-configure.libsonnet:38]: Umm, do we want to use the old name `queryCache`?

Comment (yeya24) [reply components/observatorium-configure.libsonnet:38]: Thanks, I missed that. Let me change it.

**Comment by yeya24** [comment]
@kakkoyun Hello, do I need to update the thanos image version in this pr?

**Comment by kakkoyun** [comment]
@yeya24 Yes, for the test mainly and hint the user minimum viable version for the changes.

File: tests/e2e.sh [hunk 2]
```diff
@@ -97,7 +97,7 @@ delete_cr() {
 run_test() {
 
     $KUBECTL wait --for=condition=available --timeout=10m -n observatorium-minio deploy/minio || (must_gather "$ARTIFACT_DIR" && exit 1)
-    $KUBECTL wait --for=condition=available --timeout=10m -n observatorium deploy/observatorium-xyz-thanos-query || (must_gather "$ARTIFACT_DIR" && exit 1)
+    $KUBECTL wait --for=condition=available --timeout=10m -n observatorium deploy/observatorium-xyz-thanos-query-frontend || (must_gather "$ARTIFACT_DIR" && exit 1)
```
Comment (kakkoyun) [inline tests/e2e.sh:100]: 👍

File: environments/base/default-config.libsonnet [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
   name: 'observatorium-xyz',
   namespace: 'observatorium',
-  thanosVersion: 'master-2020-05-24-079ad427',  // Fixes a blocker issue in v0.13.0-rc.0
+  thanosVersion: 'master-2020-08-12-70f89d83',  // Fixes a blocker issue in v0.13.0-rc.0
```
Comment (bwplotka) [inline environments/base/default-config.libsonnet:6]: No need for this comment anymore, right? (:

Comment (kakkoyun) [reply environments/base/default-config.libsonnet:6]: 🙈 We should clean it

File: components/observatorium.libsonnet [hunk 4]
```diff
@@ -185,19 +185,29 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
...
+      config+:: {
+        local cfg = self,
+        name: obs.config.name + '-' + cfg.commonLabels['app.kubernetes.io/name'],
+        namespace: obs.config.namespace,
+        commonLabels+:: obs.config.commonLabels,
+        replicas: 1,
+        downstreamURL: 'http://%s.%s.svc.cluster.local.:%d' % [
+          obs.query.service.metadata.name,
+          obs.query.service.metadata.namespace,
+          9090,
```
Comment (yeya24) [inline components/observatorium.libsonnet:204]: Sorry, I just found I hardcode the port. It should be `obs.query.service.spec.ports[1].port`. I will open another pr to fix it. @kakkoyun

**Pull request merged**